community/values/src/main/java/org/neo4j/values/storable/DurationValue.java

```diff
@@ -238,12 +238,14 @@ else if ( seconds > 0 && nanos < 0 )
     }
 
     @Override
-    public int compareTo( Value otherValue )
+    public int compareTo( DurationValue other )
+    {
+        return unsafeCompareTo( other );
```

Comment:
Here we should flip so that `unsafeCompareTo` calls `compareTo( (DurationValue) otherValue )`, and `compareTo` has the logic.